tempto-core/src/main/java/com/teradata/tempto/internal/query/QueryResultValueComparator.java

```diff
@@ -183,14 +184,24 @@ private int booleanEqual(Object actual, Object expected)
 
     private int longEqual(Object actual, Object expected)
     {
-        if (!(isIntegerValue(actual) && isIntegerValue(expected))) {
+        if (!(actual instanceof Number && expected instanceof Number)) {
```

Comment:
You should expect here only instances of type `Long` since you provided a separate method for other ints (`integerEqual`).